server/static/editor-plugins/addons.js

```diff
@@ -1,10 +1,99 @@
 // https://h5p.org/adding-text-editor-buttons
 
+function getFixMathTypeDialog() {
+  let offsetTop = 0;
+  let viewportHeight = 0;
+  let parentOffsetTop = 0;
+  let parentWidth = 0;
+  let parentHeight = 0;
+  let mathTypeInstances = [];
+  let waitingForUpdate = false;
+
+  const setMathTypePosition = () => {
+    const visibleInstances = mathTypeInstances.filter(
+      (editor) => !editor.classList.contains('wrs_closed')
+    );
+    if (waitingForUpdate || visibleInstances.length === 0) return;
+    waitingForUpdate = true;
+    requestAnimationFrame(() => {
+      try {
+        const boundingRects = visibleInstances
+          .map((dialog) => dialog.getBoundingClientRect())
+        const instanceHeights = boundingRects.map((rect) => rect.height)
+        const totalHeight = instanceHeights
+          .reduce((ax, x) => ax + x);
+        const startingOffsetTop =
+          offsetTop - parentOffsetTop + viewportHeight / 2 - totalHeight / 2;
+        let relOffsetTop = 0;
+        visibleInstances.forEach((dialog, idx) => {
+          const dialogWidth = boundingRects[idx].width;
+          const top = Math.min(
+            Math.max(startingOffsetTop, 0), parentHeight - totalHeight
+          ) + relOffsetTop;
+          const left = Math.max((parentWidth / 2 - dialogWidth / 2), 0);
+          relOffsetTop += instanceHeights[idx];
+          dialog.style.inset = `${top}px ${left}px 0px`;
+        });
+      } catch (e) {
+        console.error(e);
+      } finally {
+        waitingForUpdate = false;
+      }
+    });
+  }
+
+  const isMathTypeNode = (node) =>
+    node.id === 'wrs_code' || node.id?.startsWith('wrs_modal_dialogContainer')
```

Comment:
The selector below uses `.wrs_modal_dialogContainer` but this checks an ID... why use different techniques for matching the element?

Reply:
The id of each dialog is like `wrs_modal_dialogContainer[0]`, `wrs_modal_dialogContainer[1]`, etc.

I agree that it is a little confusing that the selector uses class while the function uses ID and will change this to `node.classList?.contains('wrs_modal_dialogContainer')` instead.